test/components.test.tsx

```diff
@@ -0,0 +1,114 @@
+import { StrictMode, forwardRef } from 'react';
+import { render, screen } from '@testing-library/react';
+
+import type { CheckboxFormatterProps, DataGridProps } from '../src/';
+import DataGrid, { DataGridDefaultComponentsProvider, SelectColumn } from '../src/';
+import type { Column } from '../src';
+import { getRows, setup } from './utils';
+
+interface Row {
+  id: number;
+}
+
+const columns: readonly Column<Row>[] = [
+  SelectColumn,
+  {
+    key: 'col',
+    name: 'Column'
+  }
+];
+
+function NoRowsFallback() {
+  return <div>Local no rows fallback</div>;
+}
+
+function GlobalNoRowsFallback() {
+  return <div>Global no rows fallback</div>;
+}
+
+const Checkbox = forwardRef<HTMLInputElement, CheckboxFormatterProps>(function Checkbox() {
```

Comment:
Does it not work without `forwardRef`?

Reply:
No, gives a type error

Reply:
Maybe we should allow `Maybe<ForwardRefExoticComponent<...> | ComponentType<...>>`

Reply:
I have made ref option but we should mention in docs that focus won't work if it is not set